### DISCUSSION TITLE
Why are colors hard-coded in bootstrap.css instead of using :root variables (example --bs-primary)

### DISCUSSION THREAD

**Discussion by mmarras** [twbs/bootstrap#34795] [Q&A]
I wanted to update my color-scheme without sass or anything fancy by just overwriting the `:root` variables for the color in my  own `css`. It works, but very limitedly, because many colors are still hard-coded (at 31 places instead of using the variable which is only used at `.bg-primary` and `.text-primary`)! Why is this done like that? I would expect that the variable is used every time when the primary color is used.

bootstrap v 5.1.0

**Comment by adityadees** [comment]
please add bootstrap version

**Reply by mmarras** [reply]
5.1.0

**Comment by mdo** [comment]
There are a number of issues and discussions around this, and blog posts, too. The goal is to incrementally improve CSS variable usage over time with v5's development. For example, v5.1.0 added dozens of new CSS variables. More will be coming.
